src/domain/wallets/primitives.ts

```diff
@@ -1,6 +1,20 @@
+import { InvalidWalletId } from "@domain/errors"
+
 // in the future, we may have:
 // - saving account (not redeemable instantly)
 // - borrowing account based on collateral
 export const WalletType = {
   Checking: "checking",
 } as const
+
+const UuidRegex =
+  /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+
+export const WalletIdRegex = UuidRegex
+
+export const checkedToWalletId = (walletId: string): WalletId | ValidationError => {
```

Comment:
Looks like we need to find out a general pattern on where to locate the `checkedToXxx` methods. Previously. you put them in `utils` now into `primitives`. This is inconsistent. I think best is colocated to where the `Xxx` thing is defined.